### PR TITLE
Cleanup and fixes

### PR DESCRIPTION
**Pull request by sjaeckel**
Some housekeeping and fixes

### CODE REVIEW

File: tests/mpi_test.c [hunk 1]
```diff
@@ -6,51 +6,51 @@
 static int s_radix_to_bin_test(void)
 {
    /* RADIX 16 */
-   const char *ghex = "2";
-   const char *phex = "FFFFFFFFFFFFFFFFC90FDAA22168C234C4C6628B80DC1CD129024E088A67CC74020BBEA63B139B22"
+   const char ghex[] = "2";
+   const char phex[] = "FFFFFFFFFFFFFFFFC90FDAA22168C234C4C6628B80DC1CD129024E088A67CC74020BBEA63B139B22"
```
Comment (jamuir) [inline tests/mpi_test.c:10]: nice. I also prefer `buf[]` rather than `buf*` since then you can use `sizeof(buf)-1` for the buffer length.

Comment (sjaeckel) [reply tests/mpi_test.c:10]: you meant the string length? :)

btw. one can also use `strlen()` for const strings, this is usually evaluated at compile time and not causing a call to the function

**Comment by karel-m** [comment]
The branch `cleanup-and-fixes` fails to build with cygwin + gcc-10.2.0
```
  * cc src/ciphers/camellia.o
In file included from ./src/headers/tomcrypt.h:78,
                 from ./src/headers/tomcrypt_private.h:4,
                 from src/ciphers/aes/aes.c:25:
./src/headers/tomcrypt_cfg.h:10: error: unterminated #ifndef
   10 | #ifndef TOMCRYPT_CFG_H
      |
```

**Comment by sjaeckel** [comment]
> The branch `cleanup-and-fixes` fails to build with cygwin + gcc-10.2.0

what the ... 5bfa8a3 apparently uncovered that `tomcrypt_cfg.h` was "broken" for how long exactly? :D 

... not really broken, but the include guards didn't do what they promised ...

**Review by karel-m — APPROVED** [review]

**Comment by sjaeckel** [comment]
@mkj do you maybe want to have a look as well?

**Pull request merged**